### ISSUE TITLE
Rancher Kubernetes UI not loading - v1.3.0

### ISSUE THREAD

**Issue by bscott** [rancher/rancher#7320]
**Rancher Versions:**
Server: v1.3.0
healthcheck: latest
ipsec: latest
network-services:
scheduler:
kubernetes (if applicable):

**Docker Version:**
1.12
**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 14.04
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernets
**Steps to Reproduce:**
Select Dashboard under Kubernetes Dropdown 
**Results:** 
Service Unavailable
**Expected:**
A Kubernetes UI

**Comment by tfiduccia** [comment]
Version - 1.3.0
This is broken again.  Click on the kubernetes UI button goes to service unavailable page.

**Comment by glacier3** [comment]
It likely tancher is not well support on swarm  or kubernetes.

**Comment by glacier3** [comment]
It is likely that rancher is not well support on swarm or kubernetes.

**Comment by pulberg** [comment]
Same result for me using kubectl locally with the generated config (~/.kube/config) - 

Rancher v1.3.0
OS - RHEL7.2
Host - VMware ESXi

```
kubectl get nodes
Error from server (InternalError): an error on the server ("Service Unavailable") has prevented the request from succeeding
```

**Comment by tfiduccia** [comment]
@pulberg - Even though I have the issue with the Dashboard UI button, I'm still able to get nodes using my config.  So two things:
1.  Can you run kubectl get nodes from the shell in the rancher UI and see if you get a response.
2.  Steps for me to get response from terminal was (I know you probably already know this but thought we could go through them just in case):
1. gcloud components install kubectl
2. vim ~/.kube/config (copy & page config info from UI)
3. kubectl get node (this returned my host name, status = Ready and the Age of the my host)

So finally, because you got a "service unavailable" this actually looks like the kuberentes service itself is not able to talk to your laptop.

**Comment by pulberg** [comment]
@tfiduccia -
1. Yes
```
> kubectl get nodes
NAME                   STATUS    AGE
testp07wblsb.rjf.com   Ready     1d
testp08wblsb.rjf.com   Ready     2d
testp09wblsb.rjf.com   Ready     2d
testp11wblsb.rjf.com   Ready     2d
```
2. Deleted and reinstalled kubectl, set up a new /.kube/config file.

3. 
```
kubectl get node
Error from server (InternalError): an error on the server ("Service Unavailable") has prevented the request from succeeding
```

Also, the URL in the generated config is "https://testp01wblsb.rjf.com/r/projects/1a6039/kubernetes", is it supposed to be https? not sure why that would be required, all of my stuff is in an "isolated" internal network.  If i change the URL to just http and put it in postman or a browser, i get a response of  - 

```
{
  "paths": [
    "/api",
    "/api/v1",
    "/apis",
    "/apis/apps",
    "/apis/apps/v1beta1",
    "/apis/authentication.k8s.io",
    "/apis/authentication.k8s.io/v1beta1",
    "/apis/authorization.k8s.io",
    "/apis/authorization.k8s.io/v1beta1",
    "/apis/autoscaling",
    "/apis/autoscaling/v1",
    "/apis/batch",
    "/apis/batch/v1",
    "/apis/batch/v2alpha1",
    "/apis/certificates.k8s.io",
    "/apis/certificates.k8s.io/v1alpha1",
    "/apis/extensions",
    "/apis/extensions/v1beta1",
    "/apis/policy",
    "/apis/policy/v1beta1",
    "/apis/rbac.authorization.k8s.io",
    "/apis/rbac.authorization.k8s.io/v1alpha1",
    "/apis/storage.k8s.io",
    "/apis/storage.k8s.io/v1beta1",
    "/healthz",
    "/healthz/poststarthook/bootstrap-controller",
    "/healthz/poststarthook/extensions/third-party-resources",
    "/healthz/poststarthook/rbac/bootstrap-roles",
    "/logs",
    "/metrics",
    "/swaggerapi/",
    "/ui/",
    "/version"
  ]
}
```

**Comment by vincent99** [comment]
Kubernetes only supports https so we have some magic to allow both to work over the same port.  That isn't the problem because "Service Unavailable" is an actual response you're getting back.

This is likely a different manifestation of the same problem where we can't find the right container to talk to that @wlan0 is looking into.

**Comment by pulberg** [comment]
@vincent99 ok, if i change the URL back to https i get this - 

```
kubectl get node
The connection to the server testp01wblsb.rjf.com was refused - did you specify the right host or port?
```

which implies that I cannot connect over https, which would be correct, because I don't have that access configured for the Rancher server in my env.

**Comment by vincent99** [comment]
The listening port (normally 8080) can accept either http or https connections.  You still need to send it to the appropriate port.  If it's 80 then that wouldn't appear in the URL for http (because it's the default) but is needed for https (because that implies 443).

**Comment by pulberg** [comment]
@vincent99 here is how I run the rancher server container mapping it's external port to 80 - 

`sudo docker run -d --restart=unless-stopped -p 80:8080 --name rancher-server rancher/server:latest --db-host abc123.rjf.com --db-port 3306 --db-user cattle --db-pass 'blahblah' --db-name cattle`

and here is the Host Reg URL - 

![screen shot 2017-01-07 at 7 07 30 pm](https://cloud.githubusercontent.com/assets/11069351/21746502/0776af44-d50d-11e6-9a85-78607b2a1b87.png)

So how is the URL in the generated k8s config getting set up as https://?

**Comment by Treenity** [comment]
Recreate a kube host make the UI come back for me

**Comment by galal-hussein** [comment]
The problem is fixed in Rancher v1.3.1-rc1 with kubernetes template (v1.5.1-rancher1-8), steps used to validate the bug is in https://github.com/rancher/rancher/issues/7360

**Comment by deniseschannon** [comment]
If your Dashboard is broken in v1.3.0 and you upgrade to v1.3.1, you will need to follow these steps:

If your dashboard was not working in v1.3.0, you will need to remove the `kubernetes-dashboard-xxx` pod after upgrading. You can find the pod in the **kube-system** namespace, under **Kubernetes** -> **Pods**.

**Comment by dalu** [comment]
Using rancherOS 0.9.1 and rancher/latest (v?)
3 nodes in VMs.
3 "public" IPv4 and IPv6s.
all hosts labelled correctly
no errors or warnings
http://192.168.178.153:8080/r/projects/1a5/kubernetes-dashboard:9090/#
Service Unavailable
I wouldn't know how to start debugging
```> kubectl get node
No resources found.
> kubectl get nodes
No resources found.```

**Comment by AleksandrNull** [comment]
I could confirm that have the same issue:
http://x.y.z.a:8080/r/projects/1a5/kubernetes-dashboard:9090/#

Service Unavailable

**Comment by GarbageYard** [comment]
I am using RancherOS and on top of it i have Rancher server (v1.6.6) running. Getting same issue when i try clicking on Kubernetes dashboard link

http://xx.yyy.zz.aa/r/projects/1a6633/kubernetes-dashboard:9090/#

**Comment by dalu** [comment]
Still present v1.7.7-rancher1 in both stable and latest on local 8 node test cluster.
Some nodes didn't have unique hostnames, could that be the reason why?

**Comment by superseb** [comment]
If they are identical, it will cause problems, yes. Please look through https://github.com/rancher/rancher/issues/10161 and https://github.com/rancher/rancher/issues/10096. If you are not hitting either, please create a new issue. Also for the other commenters, this is a closed issue regarding Rancher v1.3.
